### ISSUE TITLE
With scalar variant, scene parameter updates always trigger k-d tree rebuild

### ISSUE THREAD

**Issue by leroyvn** [mitsuba-renderer/mitsuba3#312]
## Summary

When using meshes with a scalar variant and the built-in k-d tree, scene parameter updates always trigger a k-d tree rebuild, even if the mesh is not modified. 

## System configuration

```
System information:

  OS: Ubuntu 22.04.1 LTS
  CPU: AMD Ryzen 9 5900X 12-Core Processor
  GPU: NVIDIA GeForce GT 730
  Python: 3.10.5 | packaged by conda-forge | (main, Jun 14 2022, 07:04:59) [GCC 10.3.0]
  NVidia driver: 470.141.03
  LLVM: 14.0.0

  Dr.Jit: 0.2.2
  Mitsuba: 3.0.2
     Is custom build? True
     Compiled with: Clang 11.1.0
     Variants:
        scalar_rgb
        scalar_spectral
        llvm_rgb
        llvm_ad_rgb
```

## Description

When using meshes with a scalar variant and the built-in k-d tree, scene parameter updates always trigger a k-d tree rebuild, even if the mesh is not modified. Is this behaviour intentional?

## Steps to reproduce

<details>
<summary><b>Minimal reproducer</b> <code>rebuild.py</code></summary>

```python
import itertools
import mitsuba as mi

mi.set_variant("scalar_rgb")
mi.Thread.thread().logger().set_log_level(mi.LogLevel.Info)
scene_dict = {
    "type": "scene",
    "integrator": {"type": "path"},
    "sensor": {"type": "perspective"},
}


def render_loop(mode, shape, variant, n=2):
    mi.set_variant(variant)
    mi.Log(mi.LogLevel.Warn, f"render_loop('{mode}', '{variant}', '{shape}')")

    if mode == "rebuild":
        for i in range(n):
            mi.Log(mi.LogLevel.Warn, str(i))
            scene = mi.load_dict({**scene_dict, **{"shape": {"type": shape}}})
            mi.render(scene, spp=1)

    elif mode == "update":
        scene = mi.load_dict({**scene_dict, **{"shape": {"type": shape}}})

        for i in range(n):
            mi.Log(mi.LogLevel.Warn, str(i))
            params = mi.traverse(scene)
            params.update()
            mi.render(scene, spp=1)

    else:
        raise RuntimeError


for i, (mode, shape, variant) in enumerate(
    itertools.product(
        ("rebuild", "update"),
        ("cube", "sphere"),
        ("scalar_rgb", "llvm_ad_rgb"),
    )
):
    if i != 0:
        print("=" * 80)
    render_loop(mode, shape, variant)

```
</details>

Output analysis:

Mode | Variant | Shape | Rebuilds? | Expected?
--------|-----------|----------|--------------|-------------
`rebuild` | `scalar_rgb` | `cube` | Y | ✔️
`rebuild` | `llvm_ad_rgb` | `cube` | Y | ✔️
`rebuild` | `scalar_rgb` | `sphere` | Y | ✔️
`rebuild` | `llvm_ad_rgb` | `sphere` | Y | ✔️
`update` | `scalar_rgb` | `cube` | Y | ❌
`update` | `llvm_ad_rgb` | `cube` | N | ✔️
`update` | `scalar_rgb` | `sphere` | N | ✔️
`update` | `llvm_ad_rgb` | `sphere` | N | ✔️

**Comment by Speierers** [comment]
Thanks for reporting this @leroyvn. It should be fixed now in [f57e741](https://github.com/mitsuba-renderer/mitsuba3/commit/f57e7416ac263445e1b74eeaf661361f4ba94855)

**Comment by leroyvn** [comment]
Thank you @Speierers!
